### PR TITLE
Automate Sparkling Water version detection

### PR DESCRIPTION
**Pull request by jakubhava**
This PR does several things:
 -> Cleans up the code, separates the checking for correct versions of SW and Spark and the dependency specification itself
-> It automates the H2O version detection. We can obtain it from the build_info file available on S3 for the desired Sparkling Water version
-> Updates the README.md
It makes sure that the user can:
 - either not specify anything -> then the latest SW is used for current Spark
 - specify just sw version -> then we obtain sw version and check if the correct spark is available
 - specify sw version and sw jar -> we just check if the correct spark is available

-> Automates h2o_release_table. It is now fetched from the provided pages.

The release table file needs to be generated by calling `./before_release/prepare_data_files.sh` before every release. Can we do this @navdeep-G? I'm not exactly sure what the release process for RSparkling is. This method generates the release table and attaches it to the R package.

This change has one drawback -> if the does not specify SW version, the we will use the latest version on the generated release_table file. That means that for full consistency, we should release RSparkling every time we release Sparkling Water to always fetch the last version. The user however can specify the SW version to the later one manually(For example in case new SW release is out and RSparkling is not out yet.)

However the ideal practise for me anyway is to do RSparkling release anyway after each Sparkling Water release which mitigates the tiny issue above. We need to update Readme.md anyway.

Let me know what you think @navdeep-G. This change should make managing versions always up-to-date and less error-prone. We don't need anymore to manually update the versions.

### CODE REVIEW

**Comment by jakubhava** [comment]
Our tests are passing, just need to make sure we play nicely with the R package structure(that's why the red in Travis at this stage)

File: .travis.yml [hunk 1]
```diff
@@ -8,13 +8,13 @@ jdk:
  - oraclejdk8
 
 r_github_packages:
-  - rstudio/sparklyr@2c59696
+  - rstudio/sparklyr@52461f1
```
Comment (jakubhava) [inline .travis.yml]: update to latest SparklyR with support for Spark 2.2.1

File: R/package.R [hunk 2]
```diff
@@ -5,104 +5,115 @@
...
+    check_spark_version_for_any_sw(spark_version)
+
+    # Automatically determine the latest Sparkling Water version based on current Spark version
+    sw_version = get_latest_sw_version_for_spark(spark_version)
+  }else if(!is.null(sw_version) && is.null(sw_location)){
+    # The user specified SW version, but not SW location. It means we need to fetch Sparkling Water and need to
+    # check if correct spark is available and stop if not.
+    check_spark_version_for_sw_version(spark_version, sw_version)
+  }else if(!is.null(sw_version) && !is.null(sw_location)){
+    # The user provided both SW version and SW artefact, just check if we are running on correct Spark version for the
```
Comment (navdeep-G) [inline R/package.R]: Typo: Should be `artifact`

File: R/package.R [hunk 3]
```diff
@@ -5,104 +5,115 @@
...
+    # desired Sparkling Water version
+    check_spark_version_for_sw_version(spark_version, sw_version)
+  }else{
+    # SW artefact is not empty, however Sparkling Water version is not specified.
```
Comment (navdeep-G) [inline R/package.R]: Typo: Should be `artifact`

File: R/package.R [hunk 4]
```diff
@@ -5,104 +5,115 @@
...
+    stop(sprintf("You specified location to Sparkling Artefact, but did not specify the version. The version is
```
Comment (navdeep-G) [inline R/package.R]: Typo: Should be `artifact`

File: R/package.R [hunk 5]
```diff
@@ -5,104 +5,115 @@
...
+    required argument in case Sparkling Water artefact is specified."))
```
Comment (navdeep-G) [inline R/package.R]: Typo: Should be `artifact`

File: R/package.R [hunk 6]
```diff
@@ -5,104 +5,115 @@
...
   }
-  
-  #Is a path to a Sparkling Water jar provided?
-  if ((!is.null(sw_location) && (!is.null(sw_version)) || !is.null(sw_location))) {
+
+  # If sparkling water jar artefact is specified, use it
```
Comment (navdeep-G) [inline R/package.R]: Typo: Should be `artifact`

File: .travis.yml (hunk 1 above)
Comment (navdeep-G) [reply .travis.yml]: Sounds good. We can update this to `sparklyr` without the hash once its in CRAN.

File: R/package.R (hunk 2 above)
Comment (jakubhava) [reply R/package.R]: thanks!

File: .travis.yml (hunk 1 above)
Comment (jakubhava) [reply .travis.yml]: this one is based on the officially release version, so maybe we can do just `sparklyr` now

**Comment by jakubhava** [comment]
Tests are passing, also manually verified that it works

**Comment by navdeep-G** [comment]
Changes look good. Please merge when ready.

**Comment by jakubhava** [comment]
Thanks @navdeep-G!

**Pull request merged**